### PR TITLE
Forgotten password part 1

### PR DESCRIPTION
**Pull request by emma-p**
reset-pwd-1
reset-pwd-2

- Creates a new reset password page
- When clicking the "Reset password" button, sends a reset link to the email address specified
- Redirects to the second page (email sent confirmation) if journey successful. Returns errors if not.

### CODE REVIEW

File: app/com/gu/identity/frontend/views/models/ErrorViewModel.scala [hunk 1]
```diff
@@ -25,7 +25,11 @@ object ErrorViewModel {
     "register-error-email" -> "Invalid email address; please try again.",
     "register-error-username" -> "Invalid username; your username must be between 6 and 20 characters long and contain only letters and numbers.",
     "register-error-password" -> "Invalid password; your password must be between 6 and 20 characters long.",
-    "register-error-csrf" -> "Forgery token invalid; please try again."
+    "register-error-csrf" -> "Forgery token invalid; please try again.",
+    "reset-password-error-gateway" -> "There was a problem resetting your password; please try again",
+    "reset-password-error-csrf" -> "Forgery token invalid; please try again",
+    "reset-password-error-bad-request" -> "There was a problem resetting your password; please try again.",
+    "reset-password-error-email" -> nonEmptyField("email")
```
Comment (susiecoleman) [inline app/com/gu/identity/frontend/views/models/ErrorViewModel.scala]: Would it be worth sticking a tech debt ticket in trello to move these strings out to the messages file?

File: app/com/gu/identity/frontend/views/models/ResetPasswordEmailSentViewModel.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.gu.identity.frontend.views.models
+
+import com.gu.identity.frontend.configuration.Configuration
+import com.gu.identity.frontend.models.text.ResetPasswordEmailSentText
+import play.api.i18n.Messages
+
+case class ResetPasswordEmailSentViewModel private(
```
Comment (susiecoleman) [inline app/com/gu/identity/frontend/views/models/ResetPasswordEmailSentViewModel.scala]: I think intellij has tried to help with indentation here... Should only be indented 4 spaces

Comment (emma-p) [inline app/com/gu/identity/frontend/views/models/ResetPasswordEmailSentViewModel.scala]: Thanks I hadn't noticed

**Comment by susiecoleman** [comment]
:+1: Looks good to me, nice work :-)

File: app/com/gu/identity/frontend/views/models/ErrorViewModel.scala (hunk 1 above)
Comment (emma-p) [inline app/com/gu/identity/frontend/views/models/ErrorViewModel.scala]: Done

File: public/reset-password-page.hbs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  <section class="page-content">
+      <p class="reset-password__subtitle">{{ resetPasswordText.subtitle }}</p>
+      <form id="reset-password-form" action={{ actions.reset }} method="POST">
+          <input type="hidden" name="{{ csrfToken.fieldName }}" value="{{ csrfToken.value }}">
+
+        <fieldset class="reset-password-form__fieldset">
+
+          {{#each errors}}
+              <p id="{{id}}" class="signin-form__error">{{message}}</p>
```
Comment (jamespamplin) [inline public/reset-password-page.hbs]: This class should be `reset-password__error` or something similar.

**Comment by jamespamplin** [comment]
:+1: great work!

**Pull request merged**